lib/api/core/models/repositories/repository.js

```diff
@@ -374,29 +364,55 @@ class Repository {
 
     return `repos/${this.index}/${this.collection}/${id}`;
   }
-}
 
-module.exports = Repository;
+  /**
+   * @param {object} dto
+   * @returns {Promise<ObjectConstructor>}
+   */
+  fromDTO (dto) {
+    const o = new this.ObjectConstructor();
+    o[_kuzzle] = this.kuzzle;
+    Object.assign(o, dto);
 
-/**
- * Returns a normalized search results
- * @param {object} raw - raw results from the db engine
- * @returns {object} normalized results
- */
-function formatSearchResults(raw) {
-  const result = {
-    total: 0,
-    hits: []
-  };
-
-  if (raw.scrollId) {
-    result.scrollId = raw.scrollId;
+    return Bluebird.resolve(o);
   }
 
-  if (raw.hits && raw.hits.length > 0) {
-    result.total = raw.total;
-    result.hits = raw.hits.map(document => _.assignIn({}, document._source, {_id: document._id}));
+  /**
+   * @param {ObjectConstructor} o
+   * @returns {object}
+   */
+  toDTO (o) {
+    return Object.assign({}, o);
+  }
+
+  /**
+   * Given a raw search response from ES, returns a {total: int, hits: []} object
+   * @param {object} raw
+   * @returns {Promise<object>}
+   * @private
+   */
+  _formatSearchResults (raw) {
+    const result = {
+      total: 0,
+      hits: []
+    };
+
+    if (raw.scrollId) {
+      result.scrollId = raw.scrollId;
```

Comment:
It may or may not be important, as it depends at what frequency searches will be performed, but this line of code will force crankshaft to deopt the function or prevent it to wrap the search result as a C++ class, because the result object will be identified as polymorphic.
It may be preferable to declare a null `scrollId` property in the result object above, and set it if a scroll identifier is set in the argument objects.